### ISSUE TITLE
mbed client ignores Content-Type on read request for single value resource

### ISSUE THREAD

**Issue by kmdewaal** [ARMmbed/mbed-client#509]
When a LWM2M server specifically ask for a Content-Type (i.e. TLV) the MBED client does reply with a 2.05 CONTENT in a different (ASCII/text) format.
This is not correct.

The correct behavior is specified in the LWM2M specification:
http://www.openmobilealliance.org/release/LightweightM2M/V1_0-20170208-A/OMA-TS-LightweightM2M-V1_0-20170208-A.pdf
Paragraph 8.1 on page 73 states:
If the preferred Content-Format cannot be returned then a 4.06 "Not Acceptable" value MUST be sent as a response.

Note that the Leshan LWM2M server is still easy to use with the mbed clients because of this bug. If you read a resource with the Leshan web interface, the Leshan asks for reply in TLV format. The reply from the mbed client in text format is accepted by the Leshan and so it appears to be working correct.

Preferably this bug should be fixed only after issue #508 , TLV format not supported for single value read and write, is solved by implementing supprot for TLV.

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTCLT-1857

**Comment by yogpan01** [comment]
Thanks for reporting, We will check and fix the issue.

**Comment by teetak01** [comment]
Closing the issue as inactive and due to the repository being deprecated.
 
Please take a look at our current offering at https://cloud.mbed.com/docs/current. The actively maintained version of this component can be found in http://github.com/armmbed/mbed-cloud-client.
 
Please open a new issue if this is still valid. We strongly encourage you to migrate to the latest product. New issues in this repository will most likely remain unfixed.
